### ISSUE TITLE
Existing apps can not be upgraded anymore after Rancher upgrade.

### ISSUE THREAD

**Issue by kintetsu** [rancher/rancher#32185]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
* We upgraded from Rancher 2.3.5 to 2.5.7
* After that we upgraded an existing app, just updated the used image, no change in the chart itself.

**Result:**
Deployement is stuck with error: **Wait helm template failed. Error: apiVersion 'v2' is not valid. The value must be "v1" : exit status 1**

**Other details that may be helpful:**
Our chart had already apiVersion: v2 and we did not change anything on the chart itself.
We already added a new chart catalog with helm version 3 but unfortunately we can not just delete the apps and launch them again.
Is there a possibility to upgrade the existing app to use the new chart catalog and helm3?

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): 2.5.7

**Comment by Blackclaws** [comment]
Have you tried deploying the chart with helm itself? If you export a kubeconfig from rancher and directly use helm and it works this might be classed as a rancher bug. Else you might have a faulty template that just didn't get flagged as such before. 

Where is the apiVersion: v2 coming from? I don't think there are any v2 kubernetes resources so far

**Comment by kintetsu** [comment]
We hadn't tried it with helm itself yet. We just tried to update the used image via rancher for our existing app. As a workaround we currently update the deployment via kubectl.

I will try it with helm directly and report back with results.
The apiVersion: v2 comes directly from our chart and as I understood it the problem is that Rancher uses Helm 2 instead of Helm 3 and expects apiVersion v1. It seems to be that this check was not in place with the old version of Rancher we used to first deploy the app.
